lib/brainstem/presenter_collection.rb

```diff
@@ -183,5 +191,11 @@ def check_for_old_options!(options)
         raise "PresenterCollection#presenting no longer accepts the :as option.  Use the brainstem_key annotation in your presenters instead."
       end
     end
+
+    def calculate_per_page(options)
+      per_page = [(options[:params][:per_page] || options[:per_page] || default_per_page).to_i, (options[:max_per_page] || default_max_per_page).to_i].min
+      per_page = default_per_page if per_page < 1
+      per_page
+    end
```

Comment:
This method is copied [_almost_ verbatim from the base strategy](https://github.com/mavenlink/brainstem/blob/69426421d26258451bec1d4cc743c15e6020995e/lib/brainstem/query_strategies/base_strategy.rb#L39).

I experimented with writing a `Pagination.calculate_per_page` thing, but it always ended up feeling really wrong.

Any thoughts or other ideas, @shirish-pampoorickal @bennapp @juanca ?

Reply:
It looks like all we're missing is passing in the defaults as part of the options? 

I think sharing the code will be highly important. What felt wrong about extracting into a `Pagination` module?

Reply:
In my work on sharing the code, I either had to:

- Accept a large parameter list (per_page, default_per_pax, max_per_page, default_max_per_page).
- Accept the same options hash that's used here, which has a nested `params` key that may have some stuff in it, or may not.
- Accept two params (per_page, max_per_page) and leave usages of the shared function to deal with the default value stuff.

All of these felt unfortunate

Reply:
Why are Pagination defaults defined in multiple places? Should the Pagination module be responsible for taking (just) user input and sanitizing?

Reply:
Maybe I'll leave the calculations in the strategy and pass the strategy around this file 🤔

Reply:
It's not clear to me why option 3 wasn't a good one:

> Accept two params (per_page, max_per_page) and leave usages of the shared function to deal with the default value stuff.

---

Perhaps you can elaborate on what you tried to do for each of options with code snippets? 

With more eyes, we can collaboratively reach a good abstraction or a bad one, but at least we'll have more context on it.

Reply:
Yep, makes sense @juanca. For number 3, that ends up removing most of the logic from the calculation.

Without the default and max and default max per page handling, the calculation would be:

```rb
[per_page, max_per_page].min
```

And at that point, it didn't feel necessary to have an object to do the calcuation.

Maybe this is a sign that the real problem here is deconstructing the `options`, and not performing the `per_page` calculation?